### PR TITLE
Make compilation faster include/c_types/trsp/trsp.h (Issue1929)

### LINKED ISSUE #1929: Making compilation faster  include/c_types/trsp/trsp.h
Refer to #1893 for instructions and details.

### PR DESCRIPTION
**Pull request by abhichou4**
Fixes #1929.

Changes proposed in this pull request:
- changes to `trsp.h`
- adjusting code based on changes on structure `edge_t`, `restrict_t`, and `path_element_tt`.

These changes were introduced to fix compilation errors.
-  `struct` keyword was added before respective structs.
- `#include <stdbool.h>` so bool can be used in `trsp.h`
- Moved typedef for `int_64` and `double` outside `#ifdef __cplusplus` condition cause it's needed in `trsp.c` as well.

@pgRouting/admins

### CODE REVIEW

**Comment by abhichou4** [comment]
The boost-supported version checks seem to be failing on my fork, saying `<stddef.h>` should be included in `trsp.h`. Do let me know if you think it should be included.

File: include/c_types/trsp/trsp.h [hunk 1]
```diff
@@ -32,49 +32,49 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
...
 #ifdef __cplusplus
 #   include <cstdint>
-typedef int64_t int64;
-typedef double float8;
 #else
 #   include <stdint.h>
 #endif
 
-typedef struct edge {
+typedef int64_t int64;
```
Comment (cvvergara) [inline include/c_types/trsp/trsp.h:39]: Please put this back on the `#ifdef __cplusplus`

**Review by cvvergara — CHANGES_REQUESTED** [review]

**Comment by cvvergara** [comment]
Something is wrong definitively.
as you can see by clicking the "Details" button, its not compiling
https://github.com/pgRouting/pgrouting/pull/2038/checks?check_run_id=3177663396#step:8:78
Maybe try with another one, and then come back to this one

**Comment by cvvergara** [comment]
I converted to draft, so that you can keep on working on it.
Mark as "Ready for review" when you see that all the tests pass

**Comment by cvvergara** [comment]
You also need be more explicit on the title of the PR
For example:
https://github.com/pgRouting/pgrouting/pull/2036

**Comment by abhichou4** [comment]
> Something is wrong definitively.
> as you can see by clicking the "Details" button, it's not compiling.
> https://github.com/pgRouting/pgrouting/pull/2038/checks?check_run_id=3177663396#step:8:78
> Maybe try with another one, and then come back to this one

Yes, I later noticed it fails to build. It can't seem to find `-lpostgres` while linking.

**Comment by abhichou4** [comment]
Opening another new branch to handle extensive merge conflicts.

**Pull request closed without merging**